onnx/defs/math/defs.cc

```diff
@@ -2050,15 +2050,23 @@ ONNX_OPERATOR_SET_SCHEMA(
                 auto* dim = second_shape.add_dim();
                 dim->set_dim_value(e);
               }
-            } else if (shape_input_shape.dim(0).has_dim_value()) {
-              // Attempt rank inference using shape of shape input
-              int64_t dim_value = shape_input_shape.dim(0).dim_value();
-              for (int64_t i = 0; i < dim_value; ++i) {
-                second_shape.add_dim();
-              }
+            } else {
+                const TensorShapeProto* symInput = ctx.getSymbolicInput(1);
+                if (symInput != nullptr) {
+                    // Shape inference by data propagation
+                    second_shape.CopyFrom(*symInput);
+                }  else if (shape_input_shape.dim(0).has_dim_value()) {
+                    // Attempt rank inference using shape of shape input
+                    int64_t dim_value = shape_input_shape.dim(0).dim_value();
+                    for (int64_t i = 0; i < dim_value; ++i) {
+                        second_shape.add_dim();
+                    }
+                }
             }
             bidirectionalBroadcastShapeInference(
                 input_shape, second_shape, *getOutputShape(ctx, 0));
+          } else {
+              // Use data propagation for shape inference when shape_initializer
```

Comment:
What is this else for?

Reply:
Sorry, this was just a mistake when I manually merged main